sbt-dump-event-schema/src/main/scala/org/virtuslab/ash/DumpEventSchema.scala

```diff
@@ -1,37 +1,40 @@
 package org.virtuslab.ash
 
 import better.files._
+import net.jcazevedo.moultingyaml.PimpedAny
+import net.jcazevedo.moultingyaml.YamlArray
+import net.jcazevedo.moultingyaml.YamlObject
 import sbt.Keys._
 import sbt.{File => _, _}
 import spray.json._
 
-object DumpEventSchema extends DefaultJsonProtocol {
+import org.virtuslab.ash.model._
+
+object DumpEventSchema {
+  import DumpEventSchemaJsonProtocol.typeDefinitionFormatJson
+  import DumpEventSchemaYamlProtocol._
   import DumpEventSchemaPlugin.autoImport._
 
   def apply(outputFile: File, inputDirectory: File): Unit = {
-    val jsons = for {
-      file <- inputDirectory.list.toSeq.sorted(File.Order.byName)
-    } yield new String(file.loadBytes).parseJson.asJsObject
+    val typeDefinitions = for {
+      file <- inputDirectory.list.filterNot(_.isDirectory)
+    } yield new String(file.loadBytes).parseJson.convertTo[TypeDefinition]
 
-    val filteredJsons = jsons.map { json =>
-      json.fields("annotations") match {
-        case JsArray(elements) if elements.isEmpty => JsObject(json.fields.filterNot(_._1 == "annotations"))
-        case _                                     => json
-      }
+    val filteredElements = typeDefinitions.toSeq.sortBy(_.name).toYaml match {
+      case YamlArray(elements) =>
+        elements.map { value =>
+          value.asYamlObject.fields.filter {
+            case (_, YamlArray(elements)) => elements.nonEmpty
+            case _                        => true
+          }
+        }
+      case _ => throw new IllegalStateException()
     }
-
+    val finalYaml = YamlArray(filteredElements.map(YamlObject.apply))
     for {
       writer <-
         outputFile.createIfNotExists(createParents = true).clear().newFileOutputStream().printWriter().autoClosed
-    } writer.print(JsArray(filteredJsons.toVector).toString(SchemaPrinter))
-  }
-
-  object SchemaPrinter extends SortedPrinter {
-    private val order =
-      Seq("typeSymbol", "name", "annotations", "parents", "fields", "typeName").zip(1 until 10).toMap
-    protected override def organiseMembers(members: Map[String, JsValue]): Seq[(String, JsValue)] = {
-      members.toSeq.sortBy(x => order.getOrElse(x._1, 10))
-    }
+    } writer.print(finalYaml.prettyPrint)
```

Comment:
Actually! whatever the order of attributes gonna be... is it gonna be deterministic b/w invocations?

Reply:
Ok, AFAICS is alphabetical